google-cloud-pubsub/lib/google/cloud/pubsub/subscriber/stream.rb

```diff
@@ -138,12 +137,27 @@ def modify_ack_deadline deadline, *messages
             synchronize do
               @inventory.remove mod_ack_ids
               @subscriber.buffer.modify_ack_deadline deadline, mod_ack_ids
-              unpause_streaming!
             end
 
             true
           end
 
+          ##
+          # @private
+          def release *messages
+            ack_ids = coerce_ack_ids messages
+            return if ack_ids.empty?
+
+            synchronize do
+              # Remove from inventory if the message was not explicitly acked or
+              # nacked in the callback
+              @inventory.remove ack_ids
+              # Check whether to unpause the stream only after the callback is
+              # completed and the thread is being reclaimed.
+              unpause_streaming!
```

Comment:
Do we need to check inventory fill levels before resuming? When the halts occurred I tried to call unpause strraming but it wouldn’t resume. I’ll try this PR out.